packages/clerk-js/src/ui/hooks/useScrollLock.ts

```diff
@@ -0,0 +1,82 @@
+// Reference: https://github.com/floating-ui/floating-ui/blob/c09c59d6e594c3527888a52ed0f3e8a2978663c2/packages/react/src/components/FloatingOverlay.tsx
```

Comment:
Is this copied verbatim? We need to do proper attribution if so:

```suggestion
// The following code is from Floating UI
// Source: https://github.com/floating-ui/floating-ui/blob/c09c59d6e594c3527888a52ed0f3e8a2978663c2/packages/react/src/components/FloatingOverlay.tsx
// Copyright (c) Floating UI contributors
// SPDX-License-Identifier: MIT
```